lib/data/calcdelta.js

```diff
@@ -0,0 +1,150 @@
+'use strict';
+
+var _ = require('lodash');
+
+module.exports = function calcDelta (oldData, newData) {
+
+  var delta = {'delta': true};
+  var changesFound = false;
+
+  // if there's no updates done so far, just return the full set
+  if (!oldData.sgvs) { return newData; }
+
+  function nsArrayTreatments(oldArray, newArray) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (11)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1874462189/issues/source?bid=2452374&fileBranchId=3087783#l13)